public/main.js

```diff
@@ -0,0 +1,350 @@
+  nameForm.style.visibility = "visible";
+  game.gameChoice = "playerVersusPlayer";
+};
+
+playerVersusCPU.onclick = function (event) {
+  event.preventDefault();
+  gameChoice.style.display = "none";
+  board.style.visibility = "visible";
+  playerName.textContent = game.playerTurn;
+  statusBar.style.visibility = "visible";
+  nameForm.style.display = "none";
+  game.gameChoice = "playerVersusCPU";
+};
+
+nameSubmit.onclick = function (event) {
+  event.preventDefault();
+  xName = xName.value;
+  oName = oName.value;
+  playerName.textContent = xName;
+  console.log(playerName);
+  statusBar.style.visibility = "visible";
+  nameForm.style.display = "none";
+  board.style.visibility = "visible";
+};
+
+startButton.onclick = function initialize(e) {
+  startButton.disabled = true;
+  timerCount();
+  activateGame();
+};
+
+function activateGame() {
+  game.activated = true;
+  console.log(game.gameChoice);
+  if (game.gameChoice === "playerVersusComputer") {
+    alert(`The game is afoot! It is ${game.playerTurn}'s turn!`);
+  } else if (game.gameChoice === "playerVersusPlayer") {
+    alert(`The game is afoot! It is ${xName}'s turn!`);
+  }
+  gameSetup();
+}
+
+function gameSetup() {
+  gameCells = document.getElementsByClassName("cell");
+  game.getGameState();
+  console.log(game.gameState);
+  for (let cell of gameCells) {
+    cell.addEventListener("click", gameAction);
+  }
+}
+
+function gameAction(e) {
+  game.markBoard(e);
+  console.log(game.gameChoice);
+}
+
+function timerCount() {
+  timer = setInterval(setTime, 1000);
+}
+function setTime() {
+  ++totalSec;
+  sec.innerHTML = pad(totalSec % 60);
+  min.innerHTML = pad(parseInt(totalSec / 60));
+}
+
+function pad(val) {
+  let valString = val + "";
+  if (valString.length < 2) {
+    return "0" + valString;
+  } else {
+    return valString;
+  }
+}
+
+const game = {
+  activated: false,
+  playerTurn: "X",
+  gameState: [],
+  gameChoice: "",
+  switch() {
+    if ((this.playerTurn === "X") && (this.gameChoice === "playerVersusPlayer")) {
+      this.playerTurn = "O";
+      playerName.textContent = oName;
+    } else if (
+      (this.playerTurn === "O") &&
+      (this.gameChoice === "playerVersusPlayer")
+    ) {
+      this.playerTurn = "X";
+      playerName.textContent = xName;
+    } else if (
+      (this.playerTurn === "X") &&
+      (this.gameChoice === "playerVersusCPU")
+    ) {
+      this.playerTurn = "O";
+      this.addCPUMove();
+      game.getGameState();
+      this.switch();
+    } else {
+      this.playerTurn = "X";
+      game.getGameState();
+      playerName.textContent = "O";
+      this.switch();
+    }
+  },
+  addCPUMove() {
+    // Game board marking for CPU in Player vs. Computer game
+    let randomIndex = Math.floor(Math.random() * 9);
+    console.log(randomIndex);
+    if ((this.gameState[randomIndex] !== "X") && (this.gameState[randomIndex] !== "O")) {
+      console.log(this.gameState[randomIndex]);
+      this.gameState[randomIndex] = "O";
```

Comment:
For this part here we need to create something so that the randomIndex can write to the specific cell on the board. One way could be by changing the textContent of the cell at that random index